### PR TITLE
Remove inconsistent spacing in Sponge project naming

### PR DESCRIPTION
**Pull request by 12awsomeman34**
Feedback welcomed

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/.

Here are some links to the pages that were modified:

- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/about/faq.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/about/future.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/about/glossary.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/about/history.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/about/introduction.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/about/license.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/about/tos.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/contributing/implementation/git-implementation.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/contributing/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/contributing/porting.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/contributing/versioning.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/ore/guidelines.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/ore/publish.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/assets.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/bans.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/blocks/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/configuration/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/configuration/nodes.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/entities/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/event/listeners.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/injection.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/optional/basic.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/optional/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/plugin-class.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/practices/best.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/project/maven.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/scheduler.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/22783b8/plugin/services.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/707#issuecomment-401068056), not to the direct url to the preview._

**Comment by ST-DDT** [comment]
Tough stuff... I'll have a look later, but I'm quite bad at those.

**Comment by Dannyps** [comment]
Please have #465 in mind, as this issue has been discussed there.

File: source/about/introduction.rst [hunk 1]
```diff
@@ -15,23 +15,23 @@ trying to avoid making the same mistakes as other projects; thus:
...
 
-We have two other projects in addition to the Sponge API:
+We have two other projects in addition to the SpongeAPI:
 
 (1) **SpongeForge**, a coremod for Minecraft Forge, which is an existing Minecraft: Java Edition modding framework
     famous for spurring the Minecraft modding scene. Forge lacks a cross-version API, and this is where Sponge steps
     in. Sponge allows server owners to deploy Sponge plugins with ease, making server management easier.
 
-(2) **SpongeVanilla**, a stand-alone implementation of the Sponge API, running on top of the vanilla Minecraft server.
+(2) **SpongeVanilla**, a stand-alone implementation of the SpongeAPI, running on top of the vanilla Minecraft server.
```
Comment (ST-DDT) [inline source/about/introduction.rst]: Not sure on this one.

File: source/plugin/index.rst [hunk 2]
```diff
@@ -2,9 +2,9 @@
 Creating a Plugin
 =================
 
-This section is intended for developers who wish to develop plugins with the Sponge API. The articles cover various
-aspects of the Sponge API and the concepts behind it. It is important to understand that the intent of this section is
-to help developers get started with the Sponge API, not to cover every concept. The
+This section is intended for developers who wish to develop plugins with the SpongeAPI. The articles cover various
```
Comment (ST-DDT) [inline source/plugin/index.rst]: Not sure on this one.

File: source/preparing/git.rst [hunk 3]
```diff
@@ -5,7 +5,7 @@ Installing Git
 Git is an open-source version control system, and helps tremendously with collaborative project development.
 
 The Sponge project, alongside thousands of other open-source projects, hosts its Git repositories on GitHub. Thus,
-Git is a crucial tool in the development of Sponge and the Sponge API.
+Git is a crucial tool in the development of Sponge and the SpongeAPI.
```
Comment (ST-DDT) [inline source/preparing/git.rst]: Not sure on this one.

File: source/server/getting-started/configuration/hocon.rst [hunk 4]
```diff
@@ -47,7 +47,7 @@ HOCON. Below are two examples of valid HOCON.
     }
 
 In practice, it is best to follow the formatting conventions of the HOCON configuration you are editing. When editing
-a HOCON configuration for Sponge or an individual plugin utilizing the Sponge API, the values are likely the only thing
+a HOCON configuration for Sponge or an individual plugin utilizing the SpongeAPI, the values are likely the only thing
```
Comment (ST-DDT) [inline source/server/getting-started/configuration/hocon.rst]: Not sure on this one.

File: source/server/getting-started/configuration/hocon.rst [hunk 5]
```diff
@@ -3,7 +3,7 @@ Introduction to HOCON
 =====================
 
 HOCON (Human-Optimized Config Object Notation) is an easy-to-use configuration format. It is used by Sponge and
-individual plugins utilizing the Sponge API to store important data, such as configuration or player data. HOCON files
+individual plugins utilizing the SpongeAPI to store important data, such as configuration or player data. HOCON files
```
Comment (ST-DDT) [inline source/server/getting-started/configuration/hocon.rst]: Not sure on this one.

File: source/server/getting-started/implementations/spongevanilla.rst [hunk 6]
```diff
@@ -2,13 +2,13 @@
 Installing SpongeVanilla
 ========================
 
-SpongeVanilla is a vanilla implementation of the Sponge API as a stand-alone server.
+SpongeVanilla is a vanilla implementation of the SpongeAPI as a stand-alone server.
```
Comment (ST-DDT) [inline source/server/getting-started/implementations/spongevanilla.rst]: Not sure on this one.

File: source/server/getting-started/implementations/spongevanilla.rst [hunk 7]
```diff
@@ -2,13 +2,13 @@
...
 
 
 Overview
 ========
 
-SpongeVanilla is an implementation of the Sponge API that is created by patching the vanilla Minecraft server. This
+SpongeVanilla is an implementation of the SpongeAPI that is created by patching the vanilla Minecraft server. This
```
Comment (ST-DDT) [inline source/server/getting-started/implementations/spongevanilla.rst]: Not sure on this one.

File: source/server/index.rst [hunk 8]
```diff
@@ -2,7 +2,7 @@
 Creating a Server
 =================
 
-The articles in this section explain how to create and run a server powered by an implementation of the Sponge API.
+The articles in this section explain how to create and run a server powered by an implementation of the SpongeAPI.
```
Comment (ST-DDT) [inline source/server/index.rst]: Not sure on this one.

File: source/server/index.rst [hunk 9]
```diff
@@ -11,7 +11,7 @@ articles contained within this section - test servers can be useful for plugin d
 .. note::
 
     We attempt to be implementation-agnostic when possible, meaning that instructions generally aren't specific to one
-    implementation of the Sponge API. Nonetheless, mistakes can happen.
+    implementation of the SpongeAPI. Nonetheless, mistakes can happen.
```
Comment (ST-DDT) [inline source/server/index.rst]: Not sure on this one.

File: source/server/management/plugins.rst [hunk 10]
```diff
@@ -57,5 +57,5 @@ If you are having problems with a plugin, consider the following things:
 * Do you run a current Java version? Sponge is built for Java 8 and is known to trigger a bug in the JRE 1.8.0_40, so
   make sure you use a version newer than that.
 * The plugin may be outdated. Is there a newer version of the plugin?
-* Does the plugin need a specific implementation and/or build of Sponge?  Some plugins may bypass the Sponge API, or
+* Does the plugin need a specific implementation and/or build of Sponge?  Some plugins may bypass the SpongeAPI, or
```
Comment (ST-DDT) [inline source/server/management/plugins.rst]: Not sure on this one.

**Review by ST-DDT — APPROVED** [review]
Well. I'm not sure?

File: source/about/faq.rst [hunk 11]
```diff
@@ -38,12 +38,12 @@ The Sponge Project currently develops two implementations which both use the Spo
     in. Sponge allows server owners to deploy Sponge plugins with ease, making server management easier. (SpongeForge
     was formerly known as Sponge, until it was renamed to avoid confusion).
 
-(2) **SpongeVanilla**, a stand-alone implementation of the Sponge API, running on top of the vanilla Minecraft server.
+(2) **SpongeVanilla**, a stand-alone implementation of the SpongeAPI, running on top of the vanilla Minecraft server.
```
Comment (dualspiral) [inline source/about/faq.rst]: SpongeAPI is a name and should not have "the" in front of it. You wouldn't say "a SpongeAPI", but you _might_ say "a Sponge API". Indeed, "SpongeAPI is a Sponge API" - Sponge being the organisation that owns or is custodian of an API.

Thus:

> **SpongeVanilla**, a stand-alone implementation of SpongeAPI, running on top of the vanilla Minecraft server.

This comment applies all the way though.

**Review by dualspiral — DISMISSED** [review]

**Comment by 12awsomeman34** [comment]
I think the old branch had some of these ^ fixed. I'll update the new one with the changes (I just ran a search + replace for these)

File: source/about/introduction.rst (hunk 1 above)
Comment (Inscrutable) [reply source/about/introduction.rst]: If we left out "the", then SpongeAPI would be more correct. As it presently stands is OK, so I don't think it needs this change.

File: source/about/faq.rst (hunk 11 above)
Comment (Inscrutable) [reply source/about/faq.rst]: Indeed. From the issue DannyPS linked, I said "SpongeAPI is the proper noun denoting the actual API that SpongePowered make". There was further discussion [here](https://github.com/SpongePowered/SpongeDocs/issues/437#issuecomment-328901973) which is where this issue originally took root.

File: source/about/introduction.rst (hunk 1 above)
Comment (Inscrutable) [reply source/about/introduction.rst]: ... However, if we're going to be consistent, all-or-nothing. SpongeAPI all the way.

**Comment by 12awsomeman34** [comment]
So I took an all-or-nothing approach and search-replaced any references to `the SpongeAPI`. I'm not too sure of any places where _explicitly_ using `the Sponge API` is appropriate (a short skim and I can't find any that imo wouldn't still sound just as right as `SpongeAPI`). Feedback on this appreciated

File: source/plugin/assets.rst [hunk 12]
```diff
@@ -57,5 +57,5 @@ following code:
 
     Developers coming from Bukkit or some other Java background might be familiar with the ``getResource`` and
     ``getResourceAsStream`` methods in ``Class``\es and ``ClassLoader``\s. **These methods should generally be avoided**
-    within the SpongeAPI environment in favor of the ``AssetManager`` in order to provide a more confluent way of
```
Comment (Inscrutable) [inline source/plugin/assets.rst]: This one probably still needs "the", as it's a different noun (the SpongeAPI *environment*).

File: source/plugin/buildsystem.rst [hunk 13]
```diff
@@ -54,7 +54,7 @@ IDE.
     simplify the development process for you and other people wanting to contribute to your project. This method of
     developing plugins does not receive active testing by the Sponge team.
 
-For developing plugins without a build system you need to download the SpongeAPI dependency manually from the
```
Comment (Inscrutable) [inline source/plugin/buildsystem.rst]: This one could also keep "the", as the noun is "SpongeAPI *dependency*"

File: source/plugin/project/gradle.rst [hunk 14]
```diff
@@ -95,7 +95,7 @@ Without SpongeGradle
   We recommend using :ref:`SpongeGradle <using-spongegradle>` for Gradle plugins since it will provide additional Gradle
   integration for Sponge plugins.
 
-Generally, everything necessary to compile a Sponge plugin using Gradle can be done by simply adding the SpongeAPI
+Generally, everything necessary to compile a Sponge plugin using Gradle can be done by simply adding SpongeAPI
```
Comment (Inscrutable) [inline source/plugin/project/gradle.rst]: Another one. "the" can stay. Reads oddly without it.

File: source/plugin/project/index.rst [hunk 15]
```diff
@@ -2,7 +2,7 @@
 Setting Up Your Project
 =======================
 
-Before you can start developing your Sponge plugin, you need to set up the SpongeAPI dependency in your project. If
+Before you can start developing your Sponge plugin, you need to set up SpongeAPI dependency in your project. If
```
Comment (Inscrutable) [inline source/plugin/project/index.rst]: Ditto. "the SpongeAPI dependency" is better here.

File: source/plugin/project/maven.rst [hunk 16]
```diff
@@ -2,7 +2,7 @@
 Setting Up Maven
 ================
 
-Add the following to the ``<project>`` block of your ``pom.xml`` to add the SpongeAPI dependency:
+Add the following to the ``<project>`` block of your ``pom.xml`` to add SpongeAPI dependency:
```
Comment (Inscrutable) [inline source/plugin/project/maven.rst]: Another one. Alternative version "to add SpongeAPI as a dependency".

**Review by Inscrutable — COMMENTED** [review]
Excellent, thorough, and thank you. I found very few cases where "the" can remain.

**Pull request merged**